index.test.js

```diff
@@ -605,17 +605,21 @@ test('clears empty selector after comma', () => {
 })
 
 test('moves comment with rule', () => {
-  run('a { /*B*/ /*B2*/ b {} }', '/*B*/ /*B2*/ a b {}')
+  run('a { /*B*/ /*B2*/ b {} }', 'a { /*B*/ /*B2*/ } a b {}')
```

Comment:
Is it a expected output?

```css
.block {
  /* Element comment */
  .element {
  }
}
```

will not be moved too?

Reply:
I didn't understand you.
Your code should be converted to:
```css
.block {
  /* Element comment */
}
.block .element {
}
```
It was from this logic that I transformed the test.

Reply:
> Your code should be converted to

Is it expected output?

Seems like very soon other developers will come and will open new issues asking to not move this kind of comments

Reply:
Do you really think that the comments embedded in the rules should remain outside?
If a comment is embedded in a rule, it should remain in it. If a comment is not embedded in a rule, it must be outside of it. It's logical, isn't it?

Reply:
In fact, the plugin originally assumed this logic - when I wrote a comment in the rule, it remained in it (see the original [issue](https://github.com/postcss/postcss-nested/issues/162)). But in one of the cases there was a mistake that I fixed.

Reply:
> If a comment is embedded in a rule, it should remain in it

It is our point of view, but it is just only one of many.

Other people think that comments and bound to the rules which is bellow. For many people `/* Element comment */` here should be bound to `.element` (because this comment explain something about the `.element`).

```css
.block {
  /* Element comment */
  .element {
  }
}
```

Look more realistic example:

```css
.nav {
  /* Home button */
  &_logo {
    width: 100px;
    height: 50px;
  }
}
```

To solve this issue we need to add many different examples to the tests for many different edge-cases. This is why I do not want to do this task, since it is not a simple fix but a big research.

Reply:
> To solve this issue we need to add many different examples to the tests for many different edge-cases

What other extreme cases, I wonder?
The joke is not funny. This is solved by one optional parameter in the API. If OFF, comments are saved outside the rules, if ON, inside.

Reply:
> What other extreme cases, I wonder?

This is part of research.

> This is solved by one optional parameter 

Adding option is the last way.

Can you explain your business case? Why exactly do you need to keep the comment inside the rule. Maybe we can add some detection based on the comment content on new lines around it, or next child

Reply:
Got it. The problem is that it is unique feature for mostly 1-2 people. I do not want to change behvaiour for everyone.

The suggestions:
1. You write a plugins which will run before (to hide comment to `Rule.props`) and after `postcss-nested` (to restore them).
2. We can add `keepComment` option to the `postcss-nested` but only if it will be very small to not reduce code maintaibility.

Reply:
There is another option: most often _(a matter of formatting_) there is an empty space between the properties and the nested rule. If they want to describe a rule with a comment, the comment is "pressed" to the rule from above, leaving no space.

This can be used to separate comments into those that cannot be left inside the rule and those that can.

Example:
```css
p {
  /* comment inside the rule, save as is */

  /* comment for the rule below, move with the rule */
  a {
  }
}
```

Reply:
Yes, let's use empty new line to detect it

Reply:
Okay, I'll take care of it in the next few days and let you know when the result is.